Expensify/App
# Replace growl errors by modal in VBA flow

Replacing all growls in the VBA flow by the modal.

The error message coming from the backend is being shown in the ConfirmModal now.

Errors messages associated with inputs where we don't handle highlighting yet, like checkboxes, are shown in the ConfirmModal.

Error modal generic message: 'Please double check any highlighted field and try again'

### Fixed Issues

$ https://github.com/Expensify/App/issues/4785

### Tests / QA Steps

Company Step: /bank-account/company

1. Navigate to `/bank-account/company`
2. Fill all inputs, input an invalid 'Zip code' (less than 6 numbers)
3. Click 'Save & continue'
4. Expected results: errors in inputs validated in the front (i.e. zip code, tax id, etc) should be highlighted in the form and a generic message should be shown in a modal.<img width="889" alt="Screen Shot 2021-09-01 at 3 56 23 PM" src="https://user-images.githubusercontent.com/87341702/131756105-2d9dab69-9201-4ff0-a042-ffaaa0896905.png">
5. Input a 'Zip Code' of 6 numbers (i.e. `123123`) and input an invalid phone number (i.e. `123`).
6. Click 'Save & continue'
7. Expected results: The phone number is a field validated in the backend, we will display a specific message in the error modal.<img width="889" alt="Screen Shot 2021-09-01 at 3 57 39 PM" src="https://user-images.githubusercontent.com/87341702/131756208-19788fd7-dbd3-4bda-8d1b-32708a9d2ced.png">


ACH Contract / Beneficial Owners Step: /bank-account/contract

1. Navigate to `/bank-account/contract`
2. Check `I accept the terms and conditions` and `I certify that the information provided is true and accurate`
3. Check `Somebody else owns more than 25% of XXX`, this will add a form to input a beneficial owner's information. Don't fill any of the beneficial owner data.
4. Click 'Save & continue'
5. Expected result: Highlighted invalid field (frontend validation) with error modal with generic message <img width="889" alt="Screen Shot 2021-09-01 at 4 20 07 PM" src="https://user-images.githubusercontent.com/87341702/131757790-70007c8c-7725-44ae-aea4-c9e4d0fecca0.png">

Requestor Step: /bank-account/requestor

1. Navigate to `/bank-account/requestor`
2. Fill all inputs
3. Input a date with invalid format in 'Date of birth', in example: `asd`
4. Expected result: 'Date of birth' field is highlighted with an error, we get the error modal with the generic message (this validation happened in the frontend).<img width="889" alt="Screen Shot 2021-09-01 at 4 09 26 PM" src="https://user-images.githubusercontent.com/87341702/131757046-0bdb010e-8d11-4b7c-abd5-a93fdd0d8108.png">
5. Input a date rejected by the backend in the 'Date of birth' , in example: `2020-10-10`
6. Expected result: The error modal displays a specific error about the 'Date of birth' being incorrect. No field is highlighted.<img width="889" alt="Screen Shot 2021-09-01 at 4 12 51 PM" src="https://user-images.githubusercontent.com/87341702/131757305-a35a71f8-8dd8-4909-8085-9dd716c4c21d.png">



### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
